teos/api.py

```diff
@@ -314,3 +315,48 @@ def get_appointment(self):
                 response = {"locator": locator, "status": AppointmentStatus.NOT_FOUND}
 
         return jsonify(response), rcode
+
+    def get_subscription_info(self):
+        """
+        Gives information about a given appointment state in the Watchtower.
+
+        Only a user who has registered can retrieve this information. User must provide correct signature.
+
+        Returns:
+            :obj:`str`: A json formatted dictionary containing information about the user's subscription.
+
+            Returns not found if the user is not yet registered with the watchtower.
+
+        """
+
+        # Getting the real IP if the server is behind a reverse proxy
+        remote_addr = get_remote_addr()
+
+        # Check that data type and content are correct. Abort otherwise.
+        try:
+            request_data = get_request_data_json(request)
+
+        except InvalidParameter as e:
+            self.logger.info("Received invalid get_subscription_info request", from_addr="{}".format(remote_addr))
+            return jsonify({"error": str(e), "error_code": errors.INVALID_REQUEST_FORMAT}), HTTP_BAD_REQUEST
+
+        self.logger.info("Received get_subscription_info request", from_addr="{}".format(remote_addr))
+
+        try:
+            r = self.stub.get_subscription_info(GetSubscriptionInfoRequest(signature=request_data.get("signature")))
+
+            response = {
```

Comment:
Protobuf encodes numeric values as floats when using `Struct`, so we need to do a little tweak here since the returned numeric values are always int (and float would make no sense for expiry nor remaining slots). Notice this is only relevant because you are using `GetUserResponse` for the response of the RPC call, which uses `Struct`.


```
response = intify(
    json_format.MessageToDict(r.user, including_default_value_fields=True, preserving_proto_field_name=True)
)
```

You'll need to import `intify` from `common.tools`.